### DISCUSSION TITLE
Docs: Docs claim that React memoizes fetch results, which I don't see any React docs to support that claim

### DISCUSSION THREAD

**Discussion by tehandyb** [vercel/next.js#72983] [App Router]
### What is the documentation issue?

https://nextjs.org/docs/13/app/building-your-application/caching
Here the documentation claims that React memoizes the response from the fetch function by extending it. I don't see any React documentation to support that

### Is there any context that might help us understand?

I doubt react would extend the fetch function, and have not seen docs to support that

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/13/app/building-your-application/caching

**Comment by samcx** [comment]
@tehandyb This is still true (maybe de-duped is the better term here?). It makes sense in App Router, where composability goes down to the component instead of a page, which was the case with Page Router. With the possibly of there being more components using the same fetch, it makes sense to automatically memoize on each React render.

I'm not exactly sure where this is located in the React code, but I believe it is in there.

**Reply by icyJoseph** [reply]
I don't know the current timeline, but this was merged some time ago to React:

- https://github.com/facebook/react/pull/28896

However, Next.js did/has some home-made version? https://github.com/vercel/next.js/blob/b5666ce3ed822ff053d2dc61496b9143c1408059/packages/next/src/server/lib/incremental-cache/index.ts#L268
